### ISSUE TITLE
extract prometheus metrics into a separate crate

### ISSUE THREAD

**Issue by ordian** [paritytech/polkadot#6015]
Currently, every subsystem depends on [polkadot-node-subsystem-util](https://github.com/paritytech/polkadot/tree/master/node/subsystem-util) mostly for `Metrics`, but this crate pulls rocksdb, overseer and other heavy hitters. This is very annoying if you want to depend on just one subsystem.

**Comment by mrcnski** [comment]
For `Metrics` the subsystems can just depend on `polkadot-node-metrics`, as in your PR. But, I see that most do depend on other stuff, e.g. `runtime`, `database`, or sometimes just `TimeoutExt`. Does it make sense to move these behind feature flags?

**Comment by ordian** [comment]
`polkadot-node-metrics` crate pulls in https://github.com/paritytech/polkadot/blob/e13d10c8e834eaee55ac97956f67263880ec635c/node/metrics/Cargo.toml#L15-L17

which in turn depend on rocksdb. Only for runtime metrics `logger_hook()`. Which got me thinking whether runtime metrics are actually useful in Polkadot? Especially with the removal of the native runtime.

**Comment by bkchr** [comment]
> Which got me thinking whether runtime metrics

These metrics work in wasm as well, but you need a wasm file compiled with these metrics enabled.

**Comment by ordian** [comment]
Yes, I mean this would only work with wasm override thing, but the question remains whether it has been useful so far?
